packages/amplify_datastore/ios/Classes/types/hub/FlutterHubElement.swift

```diff
@@ -46,6 +46,39 @@ public struct FlutterHubElement {
         self.lastChangedAt = hubElement.lastChangedAt
         self.deleted = hubElement.deleted
     }
+
+    init(
+        hubElement: MutationEvent,
+        modelSchemaRegistry: FlutterSchemaRegistry,
+        customTypeSchemaRegistry: FlutterSchemaRegistry,
+        modelName: String
+    ) throws {
+        do {
+            let decodedModel = try ModelRegistry.decode(modelName: modelName, from: hubElement.json) as! FlutterSerializedModel
+            model = try decodedModel.toMap(
+                modelSchemaRegistry: modelSchemaRegistry,
+                customTypeSchemaRegistry: customTypeSchemaRegistry,
+                modelName: modelName
+            )
+            self.version = hubElement.version
+            let serializedData = model["serializedData"] as! [String: Any]
+            if let value = serializedData["_deleted"] as? Bool {
+                self.deleted = value
+            } else if serializedData["_deleted"] == nil {
+                self.deleted = false
+            }
```

Comment:
```suggestion
            self.deleted = serializedData["_deleted"] as? Bool
```

Reply:
Need to assign the default value `false` when `_delete` is `null`.
Will change to

```swift
let value = serializedData["_deleted"] as? Bool ?? false;
```

Reply:
Then make the property non-null as well on the type

Reply:
I can't, as `_deleted` is defined as nullable in a model schema.

Reply:
Oh you mean the class property, yes I'll change it.